ocsp_status.go

```diff
@@ -2,33 +2,42 @@
 package main
 
 import (
-	"bytes"
 	"crypto"
+	"crypto/x509"
 	"flag"
 	"fmt"
 	"github.com/googleinterns/ocsp-response-linter/linter"
 	"github.com/googleinterns/ocsp-response-linter/ocsptools"
 	"github.com/googleinterns/ocsp-response-linter/ocsptools/helpers"
 	"golang.org/x/crypto/ocsp"
-	"log"
 	"net/http"
+	"strings"
 )
 
 // checkFromFile takes a path to an OCSP Response file and then reads, parses, and lints it
-func checkFromFile(tools ocsptools.ToolsInterface, linter linter.LinterInterface, respFile string) error {
+func checkFromFile(tools ocsptools.ToolsInterface, linter linter.LinterInterface, respFile string, issuerFile string, verbose bool) error {
 	ocspResp, err := tools.ReadOCSPResp(respFile)
 	if err != nil {
 		return err
 	}
 
-	linter.LintOCSPResp(ocspResp)
+	var issuerCert *x509.Certificate // pointers initialize to nil
+
+	if issuerFile != "" {
+		issuerCert, err = tools.ParseCertificateFile(issuerFile)
+		if err != nil {
+			return fmt.Errorf("Error parsing certificate from certificate file: %w", err)
+		}
+	}
+
+	linter.LintOCSPResp(ocspResp, issuerCert, verbose)
 
 	return nil
 }
 
 // checkFromCert takes a path to an ASN.1 DER encoded certificate file and
 // constructs and sends an OCSP request then parses and lints the OCSP response
-func checkFromCert(tools ocsptools.ToolsInterface, linter linter.LinterInterface, certFile string, isPost bool, ocspURL string, dir string, hash crypto.Hash) error {
+func checkFromCert(tools ocsptools.ToolsInterface, linter linter.LinterInterface, certFile string, issuerFile string, isPost bool, ocspURL string, dir string, hash crypto.Hash, verbose bool) error {
```

Comment:
A note about changing method signatures of existing code. You don't need to make any change for this, but I think in general, when you change a method signature, you risk breaking a lot of places that call that method. So if it is possible to set default values for the new parameters, it is better to put them at the end so that existing code that doesn't care about those new parameters won't need to define them.
For example if i had the following python code
```
def Foo(a,b):
  return a+b
```
and I wanted to add a third value that can default to 1. I would do it like this so that other places that call Foo don't necessarily need to change. 
```
def Foo(a,b,c=1):
  return c*(a+b)
```
If I put an undefaulted `c` in between `a` and `b` then everywhere that calls it will now have the variables in the wrong order which is a lot more dangerous. Especially if you miss somewhere where the method is called.

Reply:
So Golang doesn't support default parameter values unfortunately: https://stackoverflow.com/questions/19612449/default-value-in-gos-method.

It definitely would have made my life easier!